README.rst

```diff
@@ -25,45 +25,44 @@ What is this plugin?
 
 A Perseus renderer wrapper for Kolibri that can track learning progress and save to the database.
 
-How to publish to PyPi?
+How can I install this plugin?
 ------------------------------
 
-1. Download this repo.
-2. Terminal move to the root level of repo dir and run the following commands:
-
-    ``git submodule update --init --recursive``
-
-    ``./update_perseus.sh``
-
-3. update `setup.py` to a newer version.
-4. Terminal move to `exercise_perseus_render` folder and run ``npm install``.
-5. Terminal move to the root level of repo dir and run the following commands to publish to PyPi:
+1. Inside your Kolibri virtual environment:
+    ``pip install kolibri-perseus-exercise-plugin``
 
-    ``python setup.py register -r pypi``
+2. Activate the plugin:
 
-    ``python setup.py sdist upload -r pypi``
+    ``kolibri plugin exercise_perseus_renderer enable``
 
+3. Restart Kolibri.
 
-How can I install this plugin?
+How can I install this plugin for development?
 ------------------------------
 
 1. Download this repo.
-2. Terminal move to the repo dir and run the following commands:
+2. Open terminal in your Kolibri repo.
+4. run the following commands:
 
-    ``git submodule update --init --recursive``
+    ``pip install -e <KOLIBRI-PERSEUS-PLUGIN-LOCAL-PATH>``
 
-    ``./update_perseus.sh``
+    ``kolibri plugin exercise_perseus_renderer enable``
 
-3. Terminal move to your Kolibri dir
-4. run the following commands:
+5. Then run the commands to install frontend packages in Kolibri, this plugin will have its recursively installed.
```

Comment:
I was able to run this eventually after restarting. (It was hanging earlier - not sure why).

Now it installs completely, but I get a 404 on mathjax:

```bash
(le)~/P/l/r/kolibri (brand) $ yarn install
yarn install v0.19.1
$ node ./frontend_build/src/npm_deprecation_warning.js
[1/4] 🔍  Resolving packages...
success Already up-to-date.
$ node ./frontend_build/src/install_dependencies.js
INFO     Writing webpack_json output to /var/folders/n9/k997qvqs3f7_m_310crwsxyc0000gp/T/117112-2206-3uo5ja.5zicm78pvi.json
INFO:kolibri.core.webpack.management.commands.webpack_json:Writing webpack_json output to /var/folders/n9/k997qvqs3f7_m_310crwsxyc0000gp/T/117112-2206-3uo5ja.5zicm78pvi.json
Installing kolibri/plugins/coach_tools/package.json...
yarn install v0.19.1
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.11s.

Installing kolibri/plugins/document_pdf_render/package.json...
yarn install v0.19.1
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.32s.

Installing kolibri/plugins/audio_mp3_render/package.json...
yarn install v0.19.1
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.31s.

Installing /Users/d/Projects/learning_equality/repos/kolibri-exercise-perseus-plugin/exercise_perseus_renderer/package.json...
yarn install v0.19.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
$ git submodule update --init --recursive && cd node_modules/perseus && yarn install
Submodule path 'node_modules/perseus': checked out '55e99dd019440ef72418fc74321111ddbadedf74'
Submodule path 'node_modules/perseus/hubble': checked out '2f376e8f9fd0a67d03c047c5057c402fba6447e5'
Submodule path 'node_modules/perseus/kmath': checked out '92454b19c2ca7f8c54f61a6ff5c2d7789cef9631'
Submodule path 'node_modules/perseus/lib/mathjax': checked out 'ae516f597892a400abff79b73cecb460ac18501b'
Submodule path 'node_modules/perseus/math-input': checked out '2f4a70ffb1acd41845d5f0f39fae33da2bab33e9'
Submodule path 'node_modules/perseus/react-components': checked out '777fd2ff24fa0af33237d5b28039729534155c23'
Submodule path 'node_modules/perseus/react-components/react-live-editor': checked out 'a050794677ea55f904ff4b6ed9d646223975750d'
Submodule path 'node_modules/perseus/simple-markdown': checked out '27d1bc0a3efecd34a35f0da0659db50dcc4f49fc'
yarn install v0.19.1
info No lockfile found.
warning perseus@0.0.1: "dependencies" has dependency "aphrodite" with range "^0.3.0" that collides with a dependency in "devDependencies" of the same name with version "^0.3.1"
[1/4] Resolving packages...
warning jsx-loader > react-tools@0.11.2: react-tools is deprecated. For more information, visit https://fb.me/react-tools-deprecated
warning jsx-loader > react-tools > commoner > graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
warning mocha > jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
warning mocha > glob > minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning mocha > glob > graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
warning jsx-loader > react-tools > commoner > glob > minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning less > request > node-uuid@1.4.7: use uuid module instead
warning jest-cli > istanbul > fileset > minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning webpack > watchpack > chokidar > readdirp > minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "draft-js@0.7.0" has unmet peer dependency "react@^0.14.0 || ^15.0.0-rc".
warning "draft-js@0.7.0" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0-rc".
[4/4] Building fresh packages...
success Saved lockfile.
Done in 26.85s.
Done in 34.61s.

Installing kolibri/plugins/video_mp4_render/package.json...
yarn install v0.19.1
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.41s.

✨  Done in 42.39s.
(le)~/P/l/r/kolibri (brand) $ 
(le)~/P/l/r/kolibri (brand) $ 
(le)~/P/l/r/kolibri (brand) $ 
kolibri manage devserver --debug -- --webpack
Traceback (most recent call last):
  File "/Users/d/PythonEnvs/le/bin/kolibri", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2927, in <module>
    @_call_aside
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2913, in _call_aside
    f(*args, **kwargs)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2940, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/pkg_resources/__init__.py", line 635, in _build_master
    ws.require(__requires__)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/pkg_resources/__init__.py", line 943, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/pkg_resources/__init__.py", line 829, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'kolibri-exercise-perseus-plugin==0.3.2' distribution was not found and is required by kolibri
(le)~/P/l/r/kolibri (brand) $ 
pip install -e ../kolibri-exercise-perseus-plugin/
Obtaining file:///Users/d/Projects/learning_equality/repos/kolibri-exercise-perseus-plugin
Installing collected packages: kolibri-exercise-perseus-plugin
  Found existing installation: kolibri-exercise-perseus-plugin 0.3.2
    Uninstalling kolibri-exercise-perseus-plugin-0.3.2:
      Successfully uninstalled kolibri-exercise-perseus-plugin-0.3.2
  Running setup.py develop for kolibri-exercise-perseus-plugin
Successfully installed kolibri-exercise-perseus-plugin
(le)~/P/l/r/kolibri (brand) $ 
kolibri manage devserver --debug -- --webpack
DEBUG    Debug mode is on!
INFO     Starting webpack process from Django runserver command
INFO     Django Runserver command has spawned a Webpack watcher process on pid 3344
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, auth, content, contenttypes, django_q, logger
Running migrations:
  No migrations to apply.
yarn run v0.19.1
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
$ node ./frontend_build/src/webpackdevserver.js 
DEBUG    Debug mode is on!
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, auth, content, contenttypes, django_q, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
Performing system checks...

System check identified no issues (1 silenced).
February 13, 2017 - 03:45:56
Django version 1.9.7, using settings 'kolibri.deployment.default.settings.base'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
INFO     Writing webpack_json output to /var/folders/n9/k997qvqs3f7_m_310crwsxyc0000gp/T/117113-3346-wbylsb.osf7ogk3xr.json
INFO:kolibri.core.webpack.management.commands.webpack_json:Writing webpack_json output to /var/folders/n9/k997qvqs3f7_m_310crwsxyc0000gp/T/117113-3346-wbylsb.osf7ogk3xr.json
Hash: b6b1e7364ada7d0b25632d3e06b28e3dc416be5d3176fce64252cf82c8d5dd14e15bfd3ab747f24e402b8323afad502163d172e231d79fbef7bae0a85e4c0de430ac06aa7dc267c644fd275389e4ccf999ce57f5267c53b91ffa
Version: webpack 1.13.1
Child default_frontend:
    Hash: b6b1e7364ada7d0b2563
    Version: webpack 1.13.1
    Time: 32989ms
                                                   Asset     Size  Chunks             Chunk Names
    loading-spinner.gif?e850c29286b4eb51d1e5e00738b3fc04  98.8 kB          [emitted]  
       kolibri-logo.svg?e9efee91576fa4fabe5eb6a15b52ddd9  15.7 kB          [emitted]  
                               default_frontend-0.2c1.js  5.24 MB       0  [emitted]  default_frontend
                                            1.1-0.2c1.js   478 kB       1  [emitted]  
Child document_pdf_render_module:
    Hash: 2d3e06b28e3dc416be5d
    Version: webpack 1.13.1
    Time: 14210ms
                                  Asset     Size  Chunks             Chunk Names
    document_pdf_render_module-0.2c1.js  88.8 kB       0  [emitted]  document_pdf_render_module
Child video_mp4_render_module:
    Hash: 3176fce64252cf82c8d5
    Version: webpack 1.13.1
    Time: 27957ms
                                           Asset     Size  Chunks             Chunk Names
    VideoJS.eot?00103e881a36640a08f869ff6888f0fd   5.9 kB          [emitted]  
    VideoJS.eot?9499bb60a03b7e136a699cd87b159cc0  7.25 kB          [emitted]  
                video_mp4_render_module-0.2c1.js  2.57 MB       0  [emitted]  video_mp4_render_module
    
    WARNING in kolibri/plugins/video_mp4_render/~/video.js/dist/video.js
    Critical dependencies:
    13:480-487 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
     @ kolibri/plugins/video_mp4_render/~/video.js/dist/video.js 13:480-487
Child coach_tools_module:
    Hash: dd14e15bfd3ab747f24e
    Version: webpack 1.13.1
    Time: 33346ms
                          Asset    Size  Chunks             Chunk Names
    coach_tools_module-0.2c1.js  838 kB       0  [emitted]  coach_tools_module
Child setup_wizard:
    Hash: 402b8323afad502163d1
    Version: webpack 1.13.1
    Time: 25981ms
                                            Asset     Size  Chunks             Chunk Names
    logo-min.png?eb2802030a74ce18b835e13478e6c62e  13.4 kB          [emitted]  
                            setup_wizard-0.2c1.js  79.9 kB       0  [emitted]  setup_wizard
Child management_module:
    Hash: 72e231d79fbef7bae0a8
    Version: webpack 1.13.1
    Time: 29951ms
                         Asset    Size  Chunks             Chunk Names
    management_module-0.2c1.js  581 kB       0  [emitted]  management_module
Child learn_module:
    Hash: 5e4c0de430ac06aa7dc2
    Version: webpack 1.13.1
    Time: 31620ms
                    Asset    Size  Chunks             Chunk Names
    learn_module-0.2c1.js  750 kB       0  [emitted]  learn_module
Child audio_mp3_render_module:
    Hash: 67c644fd275389e4ccf9
    Version: webpack 1.13.1
    Time: 30325ms
                               Asset    Size  Chunks             Chunk Names
    audio_mp3_render_module-0.2c1.js  173 kB       0  [emitted]  audio_mp3_render_module
Child exercise_perseus_render_module:
    Hash: 99ce57f5267c53b91ffa
    Version: webpack 1.13.1
    Time: 34437ms
                                                               Asset     Size  Chunks             Chunk Names
              KaTeX_Math-Italic.eot?031026c46bba0e8de61c6efabd9b263a  41.7 kB          [emitted]  
              KaTeX_AMS-Regular.eot?672c96197e82b9e38e0feb8267cf10d8  71.7 kB          [emitted]  
             KaTeX_AMS-Regular.woff?e78f217c38267703d444fb8f3940a431  40.2 kB          [emitted]  
              KaTeX_AMS-Regular.ttf?9971d2707241bb6c19cb1141e3b35d76  71.4 kB          [emitted]  
         KaTeX_Caligraphic-Bold.eot?3c3fce5eb8b64ba21d19eaa6d40ff339  19.8 kB          [emitted]  
       KaTeX_Caligraphic-Bold.woff2?a2e0522575c10759bcf3332c5db0260f  10.6 kB          [emitted]  
        KaTeX_Caligraphic-Bold.woff?bac61997af03ef4747cd73b3757749ca  12.1 kB          [emitted]  
         KaTeX_Caligraphic-Bold.ttf?743b42a3ace84553402144b6b186eb99  19.6 kB          [emitted]  
      KaTeX_Caligraphic-Regular.eot?a0ba2817318bda0e08e59e2f07416112  19.2 kB          [emitted]  
    KaTeX_Caligraphic-Regular.woff2?479a68ec912f41b8a28b1b1f3000b889  10.4 kB          [emitted]  
     KaTeX_Caligraphic-Regular.woff?a64e134208e4b556aa9adfd286aa46ab  11.9 kB          [emitted]  
      KaTeX_Caligraphic-Regular.ttf?244db27f9e25edd519486604c233902f    19 kB          [emitted]  
             KaTeX_Fraktur-Bold.eot?2b4454d6e2f687619cbd307e75102078  36.2 kB          [emitted]  
           KaTeX_Fraktur-Bold.woff2?8e5f883e14f473f6be1dfa8c51bf277b  20.5 kB          [emitted]  
            KaTeX_Fraktur-Bold.woff?0a0aa194aa39cc284a3d8826cd23cfa9  23.4 kB          [emitted]  
             KaTeX_Fraktur-Bold.ttf?ad26cc83c0ff84ac3afc867f58848783    36 kB          [emitted]  
          KaTeX_Fraktur-Regular.eot?dc81eae9effb44e3651c8f6fd9513818  34.9 kB          [emitted]  
        KaTeX_Fraktur-Regular.woff2?ae2b6f4374a0e6ad9f7cfd75b3b68fd3  19.9 kB          [emitted]  
         KaTeX_Fraktur-Regular.woff?f980ca72a0d0876d0451f9e8d7b25c02  22.8 kB          [emitted]  
          KaTeX_Fraktur-Regular.ttf?d459632ec35250277328388039edc45d  34.7 kB          [emitted]  
                KaTeX_Main-Bold.eot?d327c2105b014c15a277a56f54927e2e  60.7 kB          [emitted]  
              KaTeX_Main-Bold.woff2?83f8b326ade537b52c54593f448e980e  29.5 kB          [emitted]  
               KaTeX_Main-Bold.woff?d8a629d21894b90448b5f42f457c2060  35.5 kB          [emitted]  
                KaTeX_Main-Bold.ttf?e69b9513bba6b6a563bad05e8a0de80c  60.5 kB          [emitted]  
              KaTeX_Main-Italic.eot?2702ac35e78692e600011ed1c85ef060  44.1 kB          [emitted]  
            KaTeX_Main-Italic.woff2?07510ed019e317ee10bb5a26012e1069    21 kB          [emitted]  
             KaTeX_Main-Italic.woff?8dd42e02d20082db960018b6488338f7  24.9 kB          [emitted]  
              KaTeX_Main-Italic.ttf?1b226149c7920e96e4c47faf0ff73d8c  43.9 kB          [emitted]  
             KaTeX_Main-Regular.eot?31ec4506469620df9b5295998a4bfead  68.2 kB          [emitted]  
           KaTeX_Main-Regular.woff2?bd65225294e9ad1114ded0a8bde4d38b  31.2 kB          [emitted]  
            KaTeX_Main-Regular.woff?2dffc87573a6d6dd440e801b5cce5c8e  37.6 kB          [emitted]  
             KaTeX_Main-Regular.ttf?d9162dfe8f36db18b8d4ef7ffa75692d    68 kB          [emitted]  
            KaTeX_AMS-Regular.woff2?f4c3270b2710ca9e0d537bea35ca5f06  33.2 kB          [emitted]  
            KaTeX_Math-Italic.woff2?afeebb76a0201bf468f5e0b1a9622c09  20.4 kB          [emitted]  
             KaTeX_Math-Italic.woff?da586018a5f1b55beb343d53bf804007  23.8 kB          [emitted]  
              KaTeX_Math-Italic.ttf?55fbb3ac09b79fc554f78b4fa51d9bf4  41.4 kB          [emitted]  
        KaTeX_SansSerif-Regular.eot?a3319b736d9c1594add4dd56dcb002bf  30.2 kB          [emitted]  
      KaTeX_SansSerif-Regular.woff2?7d5fa3e2a370f59774cd189cf6ca2e59  13.9 kB          [emitted]  
       KaTeX_SansSerif-Regular.woff?48c7df6f4d3d4df25748a666d0520b5c  16.7 kB          [emitted]  
        KaTeX_SansSerif-Regular.ttf?8075d14aa49687aad5e147668cc3402a    30 kB          [emitted]  
           KaTeX_Script-Regular.eot?cf8394e808472179776070c3aa4d8169  25.1 kB          [emitted]  
         KaTeX_Script-Regular.woff2?c472b57029cc804912e0380dcbf72491  12.3 kB          [emitted]  
          KaTeX_Script-Regular.woff?5acb381b12b66ca6afef5d9edb948672  13.9 kB          [emitted]  
           KaTeX_Script-Regular.ttf?abb12fc254db7db04627ecccf005eba6  24.9 kB          [emitted]  
            KaTeX_Size1-Regular.eot?5438d9d464a3973d450201211780eeb7  13.4 kB          [emitted]  
          KaTeX_Size1-Regular.woff2?feed6c70fde629d6eb61e1ed2bb93537  5.82 kB          [emitted]  
           KaTeX_Size1-Regular.woff?bdd0d5e034ab4a8641bd05736d5ed84f  6.98 kB          [emitted]  
            KaTeX_Size1-Regular.ttf?8cc60fd5985fceea2e8c7cbadb731eaf  13.2 kB          [emitted]  
            KaTeX_Size2-Regular.eot?1f5c2abfeea0beaf96c20fcada9ef082  12.6 kB          [emitted]  
          KaTeX_Size2-Regular.woff2?8a86a0afd5a0b172bdc93be1e3d53866  5.56 kB          [emitted]  
           KaTeX_Size2-Regular.woff?fd67fb35731da39667ae210a98c60ef4  6.68 kB          [emitted]  
            KaTeX_Size2-Regular.ttf?5976fffdaa1b8e6eed853f3dc3ba1330  12.4 kB          [emitted]  
            KaTeX_Size3-Regular.eot?1a6c0d68e798835afb71c745092d80f5   8.6 kB          [emitted]  
          KaTeX_Size3-Regular.woff2?2c1ea03043d76fc65767eb60aa6e0015  3.86 kB          [emitted]  
           KaTeX_Size3-Regular.woff?943c94f89c864bae86f603aee2bf83ea  4.78 kB          [emitted]  
            KaTeX_Size3-Regular.ttf?e929f5d97ac36718e3a66218f45925d8  8.36 kB          [emitted]  
            KaTeX_Size4-Regular.eot?5a3cee2bf91471f8b0356195afde026c  11.5 kB          [emitted]  
          KaTeX_Size4-Regular.woff2?680d35e3c073a84a91283a283e47fb0c  5.17 kB          [emitted]  
           KaTeX_Size4-Regular.woff?68537743d23b63655387918e39fe65c1  6.46 kB          [emitted]  
            KaTeX_Size4-Regular.ttf?81ab95e4bae239656034b9e6e62dcaaf  11.3 kB          [emitted]  
       KaTeX_Typewriter-Regular.eot?b2e94149f3405c883c9af607c80bf42c  35.8 kB          [emitted]  
     KaTeX_Typewriter-Regular.woff2?8a6d8ed89749165e1f09044215edb958  17.3 kB          [emitted]  
      KaTeX_Typewriter-Regular.woff?3e9e27f043e770a8690883e26599133f  20.7 kB          [emitted]  
       KaTeX_Typewriter-Regular.ttf?29017475281a903951aa79c83e38caf7  35.5 kB          [emitted]  
                             exercise_perseus_render_module-0.2c1.js  10.2 MB       0  [emitted]  exercise_perseus_render_module
    
    WARNING in .-exercise-perseus-plugin/exercise_perseus_renderer/~/perseus/lib/katex/katex.js
    Critical dependencies:
    1:478-485 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
     @ .-exercise-perseus-plugin/exercise_perseus_renderer/~/perseus/lib/katex/katex.js 1:478-485
[13/Feb/2017 03:46:40] "GET / HTTP/1.1" 302 0
[13/Feb/2017 03:46:41] "GET /learn/ HTTP/1.1" 200 56842
[13/Feb/2017 03:46:47] "GET /learn/ HTTP/1.1" 200 57776
[13/Feb/2017 03:46:49] "GET /api/content/78eed5c0b59b30c0a40c94c17c849af6/contentnode/c5a4691785ee5c96a0f6bdb308ab9c5b/ HTTP/1.1" 200 1270
[13/Feb/2017 03:46:51] "GET /static/mathjax/2.1/MathJax.js?config=KAthJax-e76365d62c31e1cbafe939bf94aa8497.js HTTP/1.1" 404 1730
```

![image](https://cloud.githubusercontent.com/assets/2367265/22870452/4084c332-f15c-11e6-9675-6ab6c66955ea.png)

![image](https://cloud.githubusercontent.com/assets/2367265/22870480/78c76128-f15c-11e6-8eca-22c6e380711b.png)